src/run/run.c

```diff
@@ -1133,18 +1133,28 @@ static int transient_service_set_properties(sd_bus_message *m, const char *pty_p
         }
 
         if (pty_path) {
-                _cleanup_close_ int pty_slave = -EBADF;
+                if (arg_transport == BUS_TRANSPORT_MACHINE)
```

Comment:
I prefer to make the function accept path + fd, so that when machined is taught to return the fd we don't need to rework the code again to pass that in.

Reply:
Can you update / replace the patch then? I'm not familiar with that part of the codebase and thing will go quicker if you apply your idea.